### ISSUE TITLE
toctree captions are missing the `caption` class on the main `p` element

### ISSUE THREAD

**Issue by ocharles** [sphinx-doc/sphinx#9051]
**Describe the bug**

Please see https://github.com/readthedocs/sphinx_rtd_theme/issues/1111, though I think I'm encountering a Sphinx bug, rather than a bug in that particular theme.

**To Reproduce**
```
<Paste your command-line here which cause the problem>

$ git clone git://github.com/circuithub/rel8
$ cd docs
$ pip install -r requirements.txt
$ make html
$ # open _build/html/index
```

**Expected behavior**

Each `toctree` caption would be a `<p class="caption">` element, which `sphinx-rtd-theme` will style specially.

**Screenshots**
Screenshots are on https://github.com/readthedocs/sphinx_rtd_theme/issues/1111

**Environment info**
- OS: Linux NixOS
- Python version: 3.8.8
- Sphinx version: 3.5.3

**Comment by tk0miya** [comment]
Good timing! I posted https://github.com/sphinx-doc/sphinx/pull/9053 to fix the trouble. Please try it if you have time :-)

**Comment by Blendify** [comment]
We have some reports on the rtd theme that unordered list are also broken.

**Comment by tk0miya** [comment]
@Blendify 

>We have some reports on the rtd theme that unordered list are also broken.

Could you explain this in detail?

**Comment by Blendify** [comment]
After looking into it, this seems not related to sphinx but our css. The issue comes from the change from `<div class="section">` to `<section>` or css selector for lists is `.section ol` we will need to update our css to fix this. The built sphinx theme may also need to be updated. I see at least an issue here: https://github.com/sphinx-doc/sphinx/blob/master/sphinx/themes/basic/static/basic.css_t#L18

**Comment by tk0miya** [comment]
@Blendify Do you have any example to reproduce the HTML output? I'd like to fix CSS, but I need to reproduce it before fix it.

**Comment by tk0miya** [comment]
I merged #9053 to the 3.5.x branch now. So the main topic of this issue is resolved in the next bugfix release.
I'll wait for a comment of Blendify for a little and will close this.

**Comment by Blendify** [comment]
Just to clarify, the issue with unordered lists is specific to the rtd-sphinx-theme and is an issue on our end.

> Do you have any example to reproduce the HTML output? I'd like to fix CSS, but I need to reproduce it before fix it.

With docutils 0.17 rst headings get <section> before 0.17 they get <section>.

Just use a simple page like:

```
*****
Title
*****

Section 1
=======

Lorem Ipsum.

Section2
=======

Lorem Ipsum.
```

To fix, check for all instances of `.section` in css files and add the same css selection but use the section tag instead of the class.

There may be more cases so we should check for all uses of new semantic tags:

`<main>, <section>, <header>, <footer>, <aside>, <figure>, and <figcaption>`

for example, I also see that `div.figure` will have to be converted to `figure`

**Comment by tk0miya** [comment]
Now I fixed the styles for `<figure>` tag on #9077. So I'm closing this now.
